airflow/providers/google/suite/operators/drive.py

```
@@ -0,0 +1,118 @@
+    :rtype: Sequence[str]
+    """
+
+    template_fields = (
+        'local_paths',
+        'drive_folder',
+    )
+
+    def __init__(
+        self,
+        local_paths: Union[Sequence[Path], Sequence[str]],
+        drive_folder: Union[Path, str],
+        gcp_conn_id: str = "google_cloud_default",
+        delete: bool = False,
+        chunk_size: int = 100 * 1024 * 1024,
+        resumable: bool = False,
+        delegate_to: Optional[str] = None,
+        impersonation_chain: Optional[Union[str, Sequence[str]]] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.local_paths = local_paths
+        self.drive_folder = drive_folder
+        self.gcp_conn_id = gcp_conn_id
+        self.delete = delete
+        self.chunk_size = chunk_size
+        self.resumable = resumable
+        self.delegate_to = delegate_to
+        self.impersonation_chain = impersonation_chain
+
+    def execute(self, context: "Context") -> Sequence[str]:
+        hook = GoogleDriveHook(
+            gcp_conn_id=self.gcp_conn_id,
+            delegate_to=self.delegate_to,
+            impersonation_chain=self.impersonation_chain,
+        )
+
+        remote_file_ids = []
+
+        for local_path in self.local_paths:
+            self.log.info(f'Uploading file to Google Drive: {local_path}')
+
+            try:
+                remote_file_id = hook.upload_file(
+                    local_location=str(local_path),
+                    remote_location=str(Path(self.drive_folder) / Path(local_path).name),
+                    chunk_size=self.chunk_size,
+                    resumable=self.resumable,
+                )
+
+                remote_file_ids.append(remote_file_id)
+
+                if self.delete:
+                    os.remove(local_path)
+                    self.log.info(f'Deleted local file: {local_path}')
+            except FileNotFoundError:
+                self.log.warning(f"File {local_path} can't be found")
```

Should this operator fail if _all_ of the files are not found? Right now the task would succeed and might give users the false sense that files were uploaded as expected. WDYT?

---

I think we gave the users all the necessary info with the return value. However, we may also discuss if we really want the operator to fail in any case. It may be useful if we give an error if no file_id has returned. But it may also have drawbacks, like having to include an `optional` parameter and such.

Also, in any case, checking for the availability of the local files shouldn't be in this operator's scope, and logging them is enough imo.

---

I think adding a parameter related to the failure behavior here would be a good idea. If this operator succeeds but all files (or maybe even if one file?) are missing and therefore not uploaded, this is a false-positive which I think will be confusing to users. With the parameter you'd give the user an explicit option to handle this scenario. I also don't necessarily think users are always going to use the output from operators or know to look at the `XComs` pushed by the task to determine what's going on.

There is an `ignore_if_missing` parameter in some places which could be a nice analog here. WDYT?

https://github.com/apache/airflow/blob/602abe8394fafe7de54df7e73af56de848cdf617/airflow/providers/microsoft/azure/operators/wasb_delete_blob.py#L52

https://github.com/apache/airflow/blob/c1ab8e2d7b68a31408e750129592e16432474512/airflow/providers/google/cloud/operators/bigquery.py#L1784

---

so the next question is: when will it fail?

When there is no file_ids returned,
or
When the returned list's length is less than the local_files' length

I think, as you said it may be misleading, the latter one makes more sense. What do you think?

---

I agree. If _any_ of the files are missing it makes sense to fail the task but configurable to allow files to be missing.